controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -405,22 +515,21 @@ public String getScopeName() {
     }
 
     private CompletableFuture<TxnStatus> checkTransactionStatus(final int epoch, final UUID txId) {
-        final CompletableFuture<TxnStatus> activeTx = getActiveTx(epoch, txId).handle((ok, ex) -> {
+
+        return verifyLegalState(() -> getActiveTx(epoch, txId).handle((ok, ex) -> {
```

Comment:
if a stream has not been created yet (say it is in creating state) and someone calls any read apis, we want to return illegal state rather than data not found exception.